src/com/ansorgit/plugins/bash/editor/EnterInStringLiteralHandler.java

```diff
@@ -74,9 +74,10 @@ public Result preprocessEnter(@NotNull PsiFile file, @NotNull Editor editor, @No
     }
 
     private PsiElement findWrappingContext(PsiElement start) {
-        if (start instanceof LeafPsiElement || start instanceof BashVar) {
-            return findWrappingContext(start.getParent());
+        PsiElement result = start;
```

Comment:
Reassigned variables and parameters is a bad practice – it's hard to read and review. 
![image](https://user-images.githubusercontent.com/426890/58287653-482fec00-7dba-11e9-8e2d-51e97051b5ac.png)
I may advise you to underline that smelly code. See reassigned variables/parameters in Java color scheme.
![image](https://user-images.githubusercontent.com/426890/58287766-83cab600-7dba-11e9-9c8a-779c64e9bd46.png)

Reply:
Okay. I'm now using `PsiTreeUtil` here.